apps/frontend/src/app/3_organisms/TransactionStepDialog/components/TransactionStep/TransactionStep.tsx

```diff
@@ -53,48 +57,43 @@ export const TransactionStep: FC<TransactionStepProps> = ({
   updateConfig,
   isLoading,
 }) => {
+  const { request } = transaction;
   const [token, setToken] = useState<TokenDetailsData | undefined>();
 
   useEffect(() => {
-    findContract(transaction.contract.address).then(result => {
-      if (result.group === 'tokens') {
-        getTokenDetailsByAddress(transaction.contract.address).then(setToken);
-      }
-    });
-  }, [transaction.contract.address]);
+    if (transaction.request.type === TxType.signTransaction) {
+      const contract = transaction.request.contract;
+      findContract(contract.address).then(result => {
+        if (result.group === 'tokens') {
+          getTokenDetailsByAddress(contract.address).then(setToken);
+        }
+      });
+    }
+  }, [transaction.request]);
 
   const { title, subtitle } = transaction;
 
   const resetConfig = useCallback(async () => {
-    try {
-      const gasLimit =
-        transaction.config?.gasLimit ??
-        (await transaction.contract.estimateGas[transaction.fnName](
-          ...transaction.args,
-        ).then(gas => gas.toString()));
+    if (request.type === TxType.signTransaction) {
```

Comment:
Extract this condition to a variable (e.g. `isSignTransaction`) - use `isTransactionRequest` function from helpers and use `useMemo`.